useHover - hook story (#1110)
mondaycom/monday-ui-react-core: src/hooks/useHover/__stories__/useHover.stories.mdx

import { Canvas, Meta, Story } from "@storybook/addon-docs";
import useHover from "../useHover";
import UsageGuidelines from "../../../storybook/components/usage-guidelines/usage-guidelines";
import "../../__stories__/general-hooks-stories.scss";

<Meta title="Hooks/useHover" />

# useHover

- [Overview](#overview)
- [Arguments](#arguments)
- [Returns](#returns)
- [Feedback](#feedback)

## Overview

> Attaches a listener for <code>mouseover</code> and <code>mouseout</code> DOM events on a specific element, returning <code>isHovered</code> state.

Reviewer: I think that this explanation is too technical and about the implementation details 
What about something like: 
```suggestion
Detect whether the mouse is hovering an element by returning is <code>isHovered</code> state.
```
feel free to change it according to what you think is best.

<Canvas>
  <Story name="Overview">
    {() => {
      const [hoverRef, isHovered] = useHover();
      return (
        <div ref={hoverRef} className="hooks-reference-element">
          Hover me! {isHovered ? "Boom!" : undefined}
        </div>
      );
    }}
  </Story>
</Canvas>

## Usage

<UsageGuidelines
  guidelines={[
    "Use this hook to get element hovered state",

>     "No arguments are needed just assign returned ref to the element"

Reviewer: ```suggestion
    "Please assign the returned hoverRef as the reference of the element"
```

  ]}
/>

## Returns

<FunctionArguments>
  <FunctionArgument
    name="ref"
    type="React.MutableRefObject"
    description={
      <>

>         A React{" "}

Reviewer: let's change to: React ref object to assign to the element which we want to track its hover state

and set the link over react ref.

by the way the link is cool!
